File: app/src/main/java/com/aurapc/admin/data/model/Promotion.java
package com.aurapc.admin.data.model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

public class Promotion implements Serializable {

    @SerializedName(value = "_id", alternate = {"id"})
    public String id;

    public String code;
    public String name;
    public String description;

    public String discountType; // percent | fixed
    public Double discountValue;
    public Double discountPercent;
    public Double maxDiscountAmount;
    public Double minOrderAmount;

    public Integer maxUsage;
    public Integer usedCount;
    public Integer maxUsagePerUser;

    public String startDate;
    public String endDate;
    public String expiresAt;

    public Boolean isActive;
    public Boolean active;

    public String createdAt;

    public boolean isExpired() {
        if (endDate != null) {
            try {
                java.util.Date d = new java.text.SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", java.util.Locale.US).parse(endDate);
                return d != null && d.getTime() < System.currentTimeMillis();
            } catch (Exception ex) {}
        }
        return false;
    }

    public boolean active() {
        if (isActive != null) return isActive;
        if (active != null) return active;
        return true;
    }
}

File: aura-backend/server/models/Promotion.js
const mongoose = require('mongoose');

const promotionSchema = new mongoose.Schema(
  {
    code: {
      type: String,
      required: true,
      unique: true,
      uppercase: true,
      trim: true,
    },
    description: { type: String, default: '' },
    discountPercent: {
      type: Number,
      required: true,
      min: 1,
      max: 100,
    },
    maxDiscountAmount: { type: Number, default: null },
    minOrderAmount: { type: Number, default: 0 },
    maxUsage: { type: Number, default: null },
    usedCount: { type: Number, default: 0 },
    maxUsagePerUser: { type: Number, default: 1 },
    startDate: { type: Date, required: true },
    endDate: { type: Date, required: true },
    isActive: { type: Boolean, default: true },
  },
  { timestamps: true }
);

promotionSchema.index({ code: 1 });
promotionSchema.index({ endDate: 1 });

module.exports = mongoose.model('Promotion', promotionSchema);
